### PR TITLE
[GraphBolt] Refactor CSCSamplingGraph and update the corresponding method.

### PR DESCRIPTION
**Pull request by keli-wen**
## Description

Since we will subsequently add a new class, `(non-fused)CSCSamplingGraph`, it would be better to refactor the current `CSCSamplingGraph` and its associated methods to facilitate the addition of the upcoming features.

However, two issues need to be addressed:

- [ ] The `BuiltinDataset` needs to be updated.
- [ ] We should determine whether to update the `SampledSubgraph` in `graphbolt/include/`. We can use `SampledSubgraph.type_per_edge == torch::nullopt` to distinguish the graph type or create a new class, such as `FusedSampledSubgraph`.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: graphbolt/src/fused_csc_sampling_graph.cc [hunk 1]
```diff
@@ -173,7 +182,7 @@ CSCSamplingGraph::GetState() const {
   return state;
 }
 
-c10::intrusive_ptr<SampledSubgraph> CSCSamplingGraph::InSubgraph(
+c10::intrusive_ptr<SampledSubgraph> FusedCSCSamplingGraph::InSubgraph(
```
Comment (frozenbugs) [inline graphbolt/src/fused_csc_sampling_graph.cc:185]: the c++ SampledSubgraph should be rename to FusedSampledSubgraph as well, let's do it in a separated PR.

**Review by frozenbugs — APPROVED** [review]

**Review by peizhou001 — APPROVED** [review]

File: docs/source/api/python/dgl.graphbolt.rst [hunk 2]
```diff
@@ -56,7 +56,7 @@ Standard Implementations
 
     OnDiskDataset
     BuiltinDataset
-    CSCSamplingGraph
+    FusedCSCSamplingGraph
```
Comment (Rhett-Ying) [inline docs/source/api/python/dgl.graphbolt.rst:59]: will we add a new wrapper or interface class such as `CSCSamplingGraph` to incorporate fused and non-fused classes?

Comment (Rhett-Ying) [reply docs/source/api/python/dgl.graphbolt.rst:59]: non-fused one will be named as `CSCSamplingGraph`

File: graphbolt/src/fused_csc_sampling_graph.cc (hunk 1 above)
Comment (keli-wen) [reply graphbolt/src/fused_csc_sampling_graph.cc:185]: Got

**Comment by keli-wen** [comment]
@dgl-bot

**Comment by keli-wen** [comment]
To ensure the CI test is successful, I have updated the `BuiltinDataset`: `test-only.zip` by regenerating it with the latest code. This may resolve the encountered `OnDiskGraphTopologyType` mismatch in the original dataset.

**Pull request merged**